docs/.vuepress/config.js

```diff
@@ -67,6 +69,17 @@ module.exports = {
       }
     }]
   ],
+  chainWebpack(config) {
```

Comment:
@LeeLenaleee I didn't follow the release of Chart.js v4: can you explain why now we need babel here?

Reply:
If I remember correctly it comes from the treeshaking fix in V4, in here we added static properties to the controller which webpack does not like without transpiling it with bable.

So without bable the docs dont work

Reply:
@stockiNail did some digging into this and it should work out of the box when you are using webpack 5 but vuepress is still working with webpack 4

Reply:
II may have missed something but I can't find any babel reference in the [main docs config](https://github.com/chartjs/Chart.js/blob/master/docs/.vuepress/config.ts#L94). I'm wondering why it's different for this project?

Reply:
That I don't know, I know that stockinail tried to solve the issue and noticed Kurkle fixed it and duplicated that fix

Reply:
@simonbrunel as reported by @LeeLenaleee, the transpiller provided out of the box in webpack (acorn) is not managing the static fields of javascript classes. This should be available in webpack version 5.35.1 but as written above, vuepress is based on version4. Why is this working in Chart.js?  That's good question. As you know, I'm not so expert :( but I think that in CHART.JS the vuepress seems to be "manage" by TS and probably it uses the TS transpiller (but maybe I'm wrong).

Reply:
Thank @stockiNail for the explanation. It looks like we will have to rely on babel indeed 😞 

@kurkle Any idea why it's different for the core docs? I'm always curious why it doesn't behave the same while apparently using the same configuration.

Reply:
> I'm always curious why it doesn't behave the same while apparently using the same configuration.

me too! I have spent time (also yesterday) and I wasn't able to find any reason. I have also tried to move the vuepress config in TS but the issue remains. But I'm going on because I need to understand ;)

Reply:
> @kurkle Any idea why it's different for the core docs? I'm always curious why it doesn't behave the same while apparently using the same configuration.

Sadly no. I did take a look, but haven't had the time to investigate properly. IMO Options are: its somehow using the umd build, or the code gets babeled for some reason (for example because its not in node_modules).

Reply:
Took a closer look, the magic seems to happen here:
https://github.com/chartjs/Chart.js/blob/784216130af7c1a15ad45774d54b34bc090b86d7/docs/.vuepress/config.ts#L94

I did not dig to the bottom, but looks like its one or a combination of: using alias, the file not being in node_modules or missing package.json in that directory - could be the trigger to use babel-loader.

Reply:
Thanks @kurkle for looking at it. I didn't think about the `node_modules` directory, which is probably the reason.